src/buildHTML.js

```diff
@@ -1473,6 +1501,230 @@ groupTypes.accent = function(group, options) {
+            options);
+        strikeThru.style.height = totalHeight + "em";
+        if (innerGroup.height <= style.metrics.xHeight) {
+            strikeThru.style.top = 0.2 + "em";
+        }
+
+        // Wrap the position:absolute span with a position:relative span.
+        node = makeSpan(["outer-wrap"], [node, strikeThru], options);
+        node.height = innerGroup.height + 0.2;
+        node.maxFontSize = innerGroup.maxFontSize;
+        node.depth = innerGroup.depth + 0.2;
+
+        // Prevent buildExpression from shifting a space into the span.
+        return makeSpan(["mord"], [new domTree.symbolNode("\u200b"), node],
+            options);
+    }
+};
+
+groupTypes.boxed = function(group, options) {
+    const innerGroup = buildGroup(group.value.body, options);
+    let node = makeSpan([], [innerGroup], options);
+
+    // Create a strut with which to enforce a span height.
+    const strut = makeSpan([options.style.reset(), "strut"], [], options);
+    strut.style.height = (innerGroup.depth + innerGroup.height + 0.528) + "em";
+    strut.style.verticalAlign = -(innerGroup.depth + 0.25) + "em";
+
+    node = makeSpan(["mord", "boxed"], [node, strut], options);
+    node.height = (innerGroup.height + 0.425);
+    node.maxFontSize = innerGroup.maxFontSize;
+    node.depth = (innerGroup.depth + 0.325);
+    if (options.color) {
+        node.style.borderColor = options.color;
+    }
+
+    // Prevent buildExpression from shifting a space into the span.
+    return makeSpan(["mord"], [new domTree.symbolNode("\u200b"), node],
+        options);
+};
+
+groupTypes.xArrow = function(group, options) {
+    const style = options.style;
+
+    // Build the argument groups in the appropriate style.
+    // Ref: amsmath.dtx:   \hbox{$\scriptstyle\mkern#3mu{#6}\mkern#4mu$}%
+
+    let newOptions = options.withStyle(style.sup());
+    const upperGroup = buildGroup(group.value.body, newOptions);
+    const upperGroupWrap = makeSpan([style.reset(), style.sup().cls()],
+        [upperGroup], newOptions);
+
+    let lowerGroup;
+    let lowerGroupWrap;
+    if (group.value.below) {
+        // Build the lower group
+        newOptions = options.withStyle(style.sub());
+        lowerGroup = buildGroup(group.value.below, newOptions);
+        lowerGroupWrap = makeSpan([style.reset(), style.sub().cls()],
+            [lowerGroup], newOptions);
+    }
+
+    const ruleWidth = fontMetrics.metrics.defaultRuleThickness /
+        style.sizeMultiplier;
+
+    const arrowBody = stretchy.svgSpan(group, options);
+
+    const arrowShift = -style.metrics.axisHeight + arrowBody.depth;
+    const upperShift = -style.metrics.axisHeight - arrowBody.height -
+        2 * ruleWidth;
+
+    // Generate the vlist
+    let vlist;
+    if (group.value.below) {
+        const lowerShift = -style.metrics.axisHeight
+            + lowerGroupWrap.height + arrowBody.height
+            + 3 * ruleWidth;
```

Comment:
I confess that this came from trial and error. I wrote this bit code before I had learned anything about how to read TeX macros. I was at the time sincerely hoping that someone would come along and put me right.

I now know more than I did and I'll take a look. I may yet need some help